usecases/case_usecase.go

```diff
@@ -762,94 +764,93 @@ func (usecase *CaseUseCase) CreateCaseFile(ctx context.Context, input models.Cre
+		if err := writer.Close(); err != nil {
 			return models.Case{}, err
 		}
 
-		err = usecase.webhookEventsUsecase.CreateWebhookEvent(ctx, tx, models.WebhookEventCreate{
-			Id:             webhookEventId,
-			OrganizationId: updatedCase.OrganizationId,
-			EventContent:   models.NewWebhookEventCaseFileCreated(updatedCase),
+		webhookEventId := uuid.New().String()
+
+		err = usecase.transactionFactory.Transaction(ctx, func(
+			tx repositories.Transaction,
+		) error {
+			if err := usecase.createCaseContributorIfNotExist(ctx, tx, input.CaseId, userId); err != nil {
+				return err
+			}
+
+			newCaseFileId := uuid.NewString()
+			if err := usecase.repository.CreateDbCaseFile(
+				ctx,
+				tx,
+				models.CreateDbCaseFileInput{
+					Id:            newCaseFileId,
+					BucketName:    usecase.caseManagerBucketUrl,
+					CaseId:        input.CaseId,
+					FileName:      fileHeader.Filename,
+					FileReference: newFileReference,
+				},
+			); err != nil {
+				return err
+			}
+
+			resourceType := models.CaseFileResourceType
+			err = usecase.repository.CreateCaseEvent(ctx, tx, models.CreateCaseEventAttributes{
+				CaseId:         input.CaseId,
+				UserId:         userId,
+				EventType:      models.CaseFileAdded,
+				ResourceType:   &resourceType,
+				ResourceId:     &newCaseFileId,
+				AdditionalNote: &fileHeader.Filename,
+			})
+			if err != nil {
+				return err
+			}
+
+			err = usecase.webhookEventsUsecase.CreateWebhookEvent(ctx, tx, models.WebhookEventCreate{
+				Id:             webhookEventId,
+				OrganizationId: creds.OrganizationId,
+				EventContent:   models.NewWebhookEventCaseFileCreated(input.CaseId),
+			})
+			if err != nil {
+				return err
+			}
+
+			return nil
 		})
 		if err != nil {
+			if deleteErr := usecase.blobRepository.DeleteFile(ctx,
+				usecase.caseManagerBucketUrl, newFileReference); deleteErr != nil {
+				logger.WarnContext(ctx, fmt.Sprintf("failed to clean up blob %s after case file creation failed", newFileReference),
+					"bucket", usecase.caseManagerBucketUrl,
+					"file_reference", newFileReference,
+					"error", deleteErr)
+				return models.Case{}, errors.Wrap(err, deleteErr.Error())
+			}
 			return models.Case{}, err
 		}
 
-		return updatedCase, nil
-	})
-	if err != nil {
-		if deleteErr := usecase.blobRepository.DeleteFile(ctx, usecase.caseManagerBucketUrl, newFileReference); deleteErr != nil {
-			logger.WarnContext(ctx, fmt.Sprintf("failed to clean up blob %s after case file creation failed", newFileReference),
-				"bucket", usecase.caseManagerBucketUrl,
-				"file_reference", newFileReference,
-				"error", deleteErr)
-			return models.Case{}, errors.Wrap(err, deleteErr.Error())
-		}
-		return models.Case{}, err
-	}
+		usecase.webhookEventsUsecase.SendWebhookEventAsync(ctx, webhookEventId)
 
-	usecase.webhookEventsUsecase.SendWebhookEventAsync(ctx, webhookEventId)
+		tracking.TrackEvent(ctx, models.AnalyticsCaseFileCreated, map[string]interface{}{
+			"case_id": input.CaseId,
+		})
```

Comment:
Same question

Reply:
same. After all the event has been committed.